### PR TITLE
Update DCOS_PACKAGE_FRAMEWORK_NAME for Chronos and HDFS

### PR DESCRIPTION
**Pull request by elingg**
Please take a look, @jsancio and @ConnorDoyle or anyone else available to review!

### CODE REVIEW

File: repo/packages/C/chronos/0/marathon.json [hunk 1]
```diff
@@ -28,6 +28,8 @@
       "portIndex": 0,
       "protocol": "HTTP"
     }
-  ]
-
+  ],
+  "labels": {
+    "DCOS_PACKAGE_FRAMEWORK_NAME": "{{chronos/mesos-framework-name}}"
```
Comment (jsancio) [inline repo/packages/C/chronos/0/marathon.json]: If we are going to do this then we need to make this a required property.

Comment (jsancio) [inline repo/packages/C/chronos/0/marathon.json]: FYI, Ideally we shouldn't expose this label stuff in the Marathon JSON. It just happen that right now we are using Marathon to store this information. This is a short term solution and very implementation specific. We should try to mimic how the Marathon package does it by using hierarchical config.json templates and a well known name.

File: repo/packages/H/hdfs/0/marathon.json [hunk 2]
```diff
@@ -15,5 +15,8 @@
       "timeoutSeconds": 20,
       "maxConsecutiveFailures": 3
     }
-  ]
+  ],
+  "labels": {
+    "DCOS_PACKAGE_FRAMEWORK_NAME": "hdfs"
```
Comment (jsancio) [inline repo/packages/H/hdfs/0/marathon.json]: See my comments in the Chronos package.

**Comment by jsancio** [comment]
When are we planning to change this package to use hierarchical config.json schema? See how the Marathon and Cassandra package do this.

File: repo/packages/C/chronos/0/marathon.json (hunk 1 above)
Comment (elingg) [inline repo/packages/C/chronos/0/marathon.json]: It has a default value actually so it's taken care of just like {{mesos/master}} and  {{chronos/zk-hosts}}. That said, I realize I need to update the command to expect that it exists. Just updated!

**Comment by elingg** [comment]
Yes, agreed, we want our services to be model citiziens. There are JIRAs open to change to the hierarchical config.json schema.

**Comment by elingg** [comment]
Please take a second look when you have a chance, @jsancio.

File: repo/packages/C/chronos/0/marathon.json (hunk 1 above)
Comment (ConnorDoyle) [inline repo/packages/C/chronos/0/marathon.json]: Please don't set these labels in the package.  These labels are managed by the DCOS package CLI.  There shouldn't be anything that specifically says "DCOS" in the universe.

Comment (ConnorDoyle) [inline repo/packages/C/chronos/0/marathon.json]: e.g. if we insist on a well-known `<package-name>.framework-name` config value for packages where `framework == true`, then the package manager can store this info as necessary.

Comment (elingg) [inline repo/packages/C/chronos/0/marathon.json]: @ConnorDoyle, I was using Cassandra as an example. It seems to be the current workaround we are using? https://github.com/mesosphere/universe/blob/master/repo/packages/C/cassandra/0/marathon.json#:L34:L36 cc @BenWhitehead

Comment (ConnorDoyle) [inline repo/packages/C/chronos/0/marathon.json]: IMO it should not be there either.

Comment (ConnorDoyle) [inline repo/packages/C/chronos/0/marathon.json]: We had an internal JIRA for how this was supposed to work (DCOS-774).  It's been closed, maybe @jsancio has more details about decisions that were made after I stopped tracking it :)

Comment (elingg) [inline repo/packages/C/chronos/0/marathon.json]: Ok, thanks Connor. I will wait until he takes a look.

Comment (jsancio) [inline repo/packages/C/chronos/0/marathon.json]: Yes, ideally Cassandra shouldn't be doing that either. Unfortunately currently Cassandra generate the framework name from other information. In this case the cluster name.

Having said that, this is not the case for Chronos. Chronos doesn't generate the framework name. It is instead configurable in a very similar fashion to Marathon. We should use the same mechanism that Marathon uses.

Comment (elingg) [inline repo/packages/C/chronos/0/marathon.json]: Let me talk to you in person, @jsancio. Chronos is like Cassandra in that it generates the framework name from other information (a config variable called mesos framework name).

**Comment by elingg** [comment]
Hello @jsancio or @ConnorDoyle, I'm pleased to use the hierarchical config schema and remove this technical debt! Now these packages can be model citizens. Please take a look.

**Comment by BenWhitehead** [comment]
Cassandra is a little different than chronos in that framework name itself isn't directly configured. What is configurable is the cassandra cluster name.

Cluster names are what has to change between cassandra clusters, as such the cassandra framework absorbs the logic to ensure that a different marathon api id and framework id are always generated for the cluster.

The reason cassandra specifies `DCOS_PACKAGE_FRAMEWORK_NAME` is so that cassandra can tell the CLI and other tools the framework name that will be registered.

File: repo/packages/C/chronos/0/config.json [hunk 3]
```diff
@@ -1,199 +1,229 @@
...
+        "type": "string"
+      }
+    }, 
+    "required": [
+      "cpus",
+      "instances",
+      "hostname",
+      "id",
+      "mem",
+      "zk-hosts"
```
Comment (jsancio) [inline repo/packages/C/chronos/0/config.json]: We should add `framework-name` as a require property.

**Comment by jsancio** [comment]
Looks good to me. One minor comment.

File: repo/packages/C/chronos/0/config.json (hunk 3 above)
Comment (elingg) [inline repo/packages/C/chronos/0/config.json]: Agreed. Done!

**Comment by elingg** [comment]
Thanks @jsancico, I made the fix based on your comment. I will merge now, unless there are any final comments.

**Comment by jsancio** [comment]
Squash the commits and merge. Thanks!

**Pull request merged**